spec/unit/provider/package/windows/registry_uninstall_entry_spec.rb

```diff
@@ -22,4 +24,40 @@
       expect(quiet_uninstall_string_key).to eql "QuietUninstallStringPath"
     end
   end
+
+  describe "when pass 'nil' to find_entries", :windows_only do
+    let(:package_name) { nil }
+    it "should return empty entries array" do
+      find_entries = Chef::Provider::Package::Windows::RegistryUninstallEntry.find_entries(package_name)
+      expect(find_entries.size).to eql 0
+    end
+  end
+
+  describe "when pass empty string to find_entries", :windows_only do
+    let(:package_name) { " " }
+
+    it "should return no entries" do
+      find_entries = Chef::Provider::Package::Windows::RegistryUninstallEntry.find_entries(package_name)
```

Comment:
`Chef::Provider::Package::Windows::RegistryUninstallEntry` is being called multiple times. It's advisable to define it at the top using `let { }`.